test/engineSpec.js

```diff
@@ -1454,69 +1454,132 @@ describe('Engine', function() {
                 done();
             });
         });
     });
 
-    describe('isNotIndependentMortgageCoOrMBS', function() {
-        it('should return true when the API response result is true', function(done) {
-            var agencyCode = '1';
-            var respondentID = '1';
-            var path = '/isNotIndependentMortgageCoOrMBS/' + engine.getRuleYear() + '/' +
-                agencyCode + '/' + respondentID;
-            mockAPI('get', path, 200, JSON.stringify({ result: true }));
-            engine.isNotIndependentMortgageCoOrMBS(respondentID, agencyCode)
+    describe('isMetroAreaOnRespondentPanel', function() {
+        it('should return true when the respondent is Independent Mortgage comapny or MBS', function(done) {
+            var hmdaFile = JSON.parse(JSON.stringify(require('./testdata/complete.json'))).hmdaFile;
+            var respondentID = '0123456789';
+            var agencyCode = '9';
+            var metroArea = '06920';
+            var path = '/isNotIndependentMortgageCoOrMBS/' + engine.getRuleYear() + '/9/0123456789';
+            mockAPI('get', path, 200, JSON.stringify({ result: false }), true);
+            
+            engine.isMetroAreaOnRespondentPanel(hmdaFile)
             .then(function(result) {
                 expect(result).to.be.true();
                 done();
             });
         });
-    });
 
-    describe('isMetroAreaOnRespondentPanel', function() {
-        it('should return true when the API response result is true', function(done) {
-            var respondentID = '1';
-            var agencyCode = '1';
-            var metroArea = '1';
-            var path = '/isMetroAreaOnRespondentPanel/' + engine.getRuleYear() + '/' +
-                agencyCode + '/' + respondentID + '/' + metroArea;
-            mockAPI('get', path, 200, JSON.stringify({ result: true }));
-            engine.isMetroAreaOnRespondentPanel(metroArea, respondentID, agencyCode)
+        it('should return true when the action taken is not valid', function(done) {
+            var hmdaFile = JSON.parse(JSON.stringify(require('./testdata/complete.json'))).hmdaFile;
+            _.each(hmdaFile.loanApplicationRegisters, function (element) {
+                element.actionTaken = '9';
+            });
+            var path = '/isNotIndependentMortgageCoOrMBS/' + engine.getRuleYear() + '/9/0123456789';
+            mockAPI('get', path, 200, JSON.stringify({ result: true }), true);
+            
+            engine.isMetroAreaOnRespondentPanel(hmdaFile)
             .then(function(result) {
                 expect(result).to.be.true();
                 done();
             });
         });
+
+        it('should return true when the respondent has a branch in the msa', function(done) {
+            var hmdaFile = JSON.parse(JSON.stringify(require('./testdata/complete.json'))).hmdaFile;
+            _.each(hmdaFile.loanApplicationRegisters, function (element) {
+                element.actionTaken = '9';
```

Comment:
Since you are setting each `actionTaken` to 9, the api for `isMetroAreaOnRespondentPanel` is never hit, so this test ends early while still returning true.